### PR TITLE
[SPARK-31697][WEBUI] HistoryServer should set Content-Type

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This PR changes HistoryServer to set Content-Type.

I noticed that we will get html as plain text when we access to wrong URLs which represent non-existence appId on HistoryServer.

```
<html>
      <head>
        <meta http-equiv="Content-type" content="text/html; charset=utf-8"/><meta name="viewport" content="width=device-width, initial-scale=1"/><link rel="stylesheet" href="/static/bootstrap.min.css" type="text/css"/><link rel="stylesheet" href="/static/vis-timeline-graph2d.min.css" type="text/css"/><link rel="stylesheet" href="/static/webui.css" type="text/css"/><link rel="stylesheet" href="/static/timeline-view.css" type="text/css"/><script src="/static/sorttable.js"></script><script src="/static/jquery-3.4.1.min.js"></script><script src="/static/vis-timeline-graph2d.min.js"></script><script src="/static/bootstrap.bundle.min.js"></script><script src="/static/initialize-tooltips.js"></script><script src="/static/table.js"></script><script src="/static/timeline-view.js"></script><script src="/static/log-view.js"></script><script src="/static/webui.js"></script><script>setUIRoot('')</script>
        
        <link rel="shortcut icon" href="/static/spark-logo-77x50px-hd.png"></link>
        <title>Not Found</title>
      </head>
      <body>
        <div class="container-fluid">
          <div class="row">
            <div class="col-12">
              <h3 style="vertical-align: middle; display: inline-block;">
                <a style="text-decoration: none" href="/">
                  <img src="/static/spark-logo-77x50px-hd.png"/>
                  <span class="version" style="margin-right: 15px;">3.1.0-SNAPSHOT</span>
                </a>
                Not Found
              </h3>
            </div>
          </div>
          <div class="row">
            <div class="col-12">
              <div class="row">Application local-1589239 not found.</div>
            </div>
          </div>
        </div>
      </body>
    </html> 
```
The reason is Content-Type not set. I confirmed it with `curl -I http://localhost:18080/history/<wrong-appId>`
```
HTTP/1.1 404 Not Found
Date: Wed, 13 May 2020 06:59:29 GMT
Cache-Control: no-cache, no-store, must-revalidate
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Content-Length: 1778
Server: Jetty(9.4.18.v20190429)
```

### Why are the changes needed?

This is a bug.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

I added a test case for this issue.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala [hunk 1]
```diff
@@ -69,6 +69,9 @@ class HistoryServer(
 
   private val loaderServlet = new HttpServlet {
     protected override def doGet(req: HttpServletRequest, res: HttpServletResponse): Unit = {
+
+      res.setContentType("text/html;charset=utf-8")
```
Comment (dongjoon-hyun) [inline core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala:73]: Thank you, @sarutak . I also confirmed this bug in 3.0.0-rc1.
BTW, 2.4.6-rc1 seems to have no problem. Do you know why we don't have this issue in 2.4.x?

Comment (sarutak) [reply core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala:73]: I think the reason is that the response header contains `X-Content-Type-Options: nosniff` for `master` and `3.0.0-rc1` while `2.4.x` doesn't.
```
HTTP/1.1 404 Not Found
Date: Wed, 13 May 2020 08:31:43 GMT
Content-Length: 1638
Server: Jetty(9.3.27.v20190418)
```

`X-Content-Type-Options: nosniff` means that the content type should be defined by `Content-Type` header, and it seems to be introduced to `3.0.x` in #23302 for security.

Comment (dongjoon-hyun) [reply core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala:73]: Got it. Thank you, @sarutak .

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. I tested this locally. Merged to master/3.0.
Thank you, @sarutak .

**Pull request closed without merging**